core/blockchain.go

```diff
@@ -1337,8 +1383,17 @@ func (bc *BlockChain) GetMaxGarbageCollectedBlockNumber() int64 {
 //
 // After insertion is done, all accumulated events will be fired.
 func (bc *BlockChain) InsertChain(chain types.Blocks, verifyHeaders bool) (int, error) {
+	// if in tikv mode, writer node need preempt master or come be a follower
+	if bc.redisPreempt != nil && !bc.tikvPreemptMaster(bc.rangeBlock(chain)) {
```

Comment:
if you put it in, the return value of the function will become unclear, I used `bc.isInitTiKV()` instead of `bc.redisPreempt != nil` to make it more readable